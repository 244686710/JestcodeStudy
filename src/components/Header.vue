<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input class="header-input" type="text" data-test="header-input" :value="inputValue"
        @input="e => changeInputVaule(e.target.value)"
        @keyup.enter="() => addTodoItem()"
        placeholder="Add UndoItem"
      >
    </div>
  </div>
</template>
<script>
import { mapState, mapMutations } from 'vuex'
export default {
  name: 'Header',

  computed: mapState(['inputValue']),
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.changeInputVaule('')
      }
    },

    ...mapMutations({
      changeInputVaule: 'changeInputValue'
    })
  }
}
</script>
<style lang="stylus" scoped>
  .header {
    background: #666;
    line-height: 66px;
  }
  .header-content {
    width 600px
    margin 0 auto
    color #333
    font-size 24px
  }
  .header-input {
    float right
    width 360px
    margin-top 16px
    line-height 24px
    color #333
    text-indent 10px
  }
</style>
