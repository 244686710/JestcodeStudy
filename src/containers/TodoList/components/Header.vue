<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input class="header-input" type="text" data-test="input" v-model="inputValue"
        @keyup.enter="addTodoItem"
        placeholder="Add UndoItem"
      >
    </div>
  </div>
</template>
<script>
export default {
  name: 'Header',
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = ''
      }
    }
  }
}
</script>
<style lang="stylus">
  .header {
    background: #666;
    line-height: 66px;
  }
  .header-content {
    width 600px
    margin 0 auto
    color #333
    font-size 24px
  }
  .header-input {
    float right
    width 360px
    margin-top 16px
    line-height 24px
    color #333
    text-indent 10px
  }
</style>
