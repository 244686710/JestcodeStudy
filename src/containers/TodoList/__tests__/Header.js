import { shallowMount } from '@vue/test-utils'
import Header from '../components/Header.vue'
import { findTestWrapper } from '../../../utils/testUtils'

describe('Header组件', () => {
  it('样式发生改变时 做出提示', () => {
    const wrapper = shallowMount(Header)
    expect(wrapper).toMatchSnapshot()
  })

  it('包含input', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    expect(input.exists()).toBe(true)
  })

  it('input 框初始内容为空', () => {
    const wrapper = shallowMount(Header)
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toBe('')
  })

  it('input值发生变化时，数据应该跟着变', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    const testValue = 'test value'
    input.setValue(testValue)
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toEqual(testValue)
  })

  it('input无内容是回车，无反应', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeFalsy()
  })

  it('input 回车有内容时，触发事件，后内容清空', () => {
    const wrapper = shallowMount(Header)
    const input = findTestWrapper(wrapper, 'input')
    input.setValue('test value')
    input.trigger('keyup.enter')
    expect(wrapper.emitted().add).toBeTruthy()
    const inputValue = wrapper.vm.$data.inputValue
    expect(inputValue).toEqual('')
  })
})
