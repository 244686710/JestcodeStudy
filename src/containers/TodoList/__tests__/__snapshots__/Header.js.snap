// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header组件 样式发生改变时 做出提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input type="text" data-test="input" placeholder="Add UndoItem" class="header-input"></div>
</div>
`;
